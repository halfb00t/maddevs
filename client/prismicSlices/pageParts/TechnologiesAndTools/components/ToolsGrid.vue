<template>
  <div class="tools-grid">
    <div
      v-for="(tool, i) in tools"
      :key="`${tool.label}-${i}`"
      :title="tool.label"
      class="tools-grid__item"
      :class="[
        `${tool.category}`,
        `${(activeCategory && tool.category === activeCategory) ? 'tools-grid__item--active' : ''}`,
        `${(activeCategory && tool.category !== activeCategory) ? 'tools-grid__item--inactive' : ''}`,
      ]"
      :style="{ backgroundColor: tool.color }"
    >
      <img
        v-lazy-load
        :data-src="$getMediaFromS3(`/images/tech-stack/${tool.icon}.svg`)"
        width="26"
        height="26"
        alt="Icon"
        class="tools-grid__item-icon"
      >
      <span class="tools-grid__item-label">
        {{ tool.label }}
      </span>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ToolsGrid',

  props: {
    activeCategory: {
      type: String,
      default: null,
    },

    tools: {
      type: Array,
      default: () => [],
    },
  },
}
</script>

<style lang="scss" scoped>
.tools-grid {
  width: calc(100% + 1px);
  margin-left: -1px;
  margin-top: -1px;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  @media screen and (max-width: 768px) {
    justify-content: flex-start;
    align-items: flex-start;
  }

  &__item {
    margin-left: 1px;
    margin-top: 1px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 102px;
    height: 90px;
    transition: opacity 0.2s linear;
    @media screen and (max-width: 768px) {
      width: 51px;
      height: 54px;
    }
    &,
    &--active {
      opacity: 1;
    }
    &--inactive {
      opacity: 0.5;
    }
    &-icon {
      display: block;
      width: 26px;
      min-width: 26px;
      height: 26px;
      @media screen and (max-width: 768px) {
        width: 20px;
        min-width: 20px;
        height: 20px;
      }
    }
    &-label {
      @include font('Inter', 12px, 400);
      display: block;
      white-space: nowrap;
      line-height: 14px;
      color: $text-color--grey-pale;
      margin-top: 11px;
      width: 82%;
      text-overflow: ellipsis;
      overflow: hidden;
      text-align: center;
      @media screen and (max-width: 768px) {
        font-size: 8px;
      }
    }
  }
}
</style>
