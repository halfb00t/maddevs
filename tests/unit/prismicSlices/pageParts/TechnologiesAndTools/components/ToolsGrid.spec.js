import { render } from '@testing-library/vue'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import lazyLoad from 'nuxt-lazy-load/lib/module'
import ToolsGrid from '@/prismicSlices/pageParts/TechnologiesAndTools/components/ToolsGrid'

const props = {
  activeCategory: 'Test category',
  tools: [
    {
      label: 'Test',
      category: 'Test category',
      icon: 'img.jpg',
    },
  ],
}

const mocks = {
  $getMediaFromS3: jest.fn(),
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('ToolsGrid component', () => {
  it('should correctly render component', () => {
    const { container } = render(ToolsGrid, {
      localVue,
      propsData: props,
      mocks,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correctly return empty array from tools props', () => {
    const wrapper = shallowMount(ToolsGrid, {
      localVue,
      mocks,
      propsData: props,
    })

    expect(wrapper.vm.$options.props.tools.default.call()).toEqual([])
  })
})
