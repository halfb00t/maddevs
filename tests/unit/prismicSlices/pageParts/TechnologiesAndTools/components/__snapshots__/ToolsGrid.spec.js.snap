// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolsGrid component should correctly render component 1`] = `
<div>
  <div
    class="tools-grid"
  >
    <div
      class="tools-grid__item Test category tools-grid__item--active"
      title="Test"
    >
      <img
        alt="Icon"
        class="tools-grid__item-icon"
        height="26"
        width="26"
      />
       
      <span
        class="tools-grid__item-label"
      >
        
      Test
    
      </span>
    </div>
  </div>
</div>
`;
